e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java

```diff
@@ -0,0 +1,512 @@
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+
+// Unit tests here can be run from the IDE and will use LOCAL as the default, they can also be run from the TestLauncher
+// class to specify a custom environment
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ExtendWith(TestRunnerParameterResolver.class)
+public class TestRunner {
+
+    private static HttpClient httpClient;
+
+    private static String AB2D_API_URL;
+
+    private static String AB2D_ADMIN_URL;
+
+    private static final String PATIENT_EXPORT_PATH = "Patient/$export";
+
+    private static String jwtStr = null;
+
+    private static String TEST_USER;
+
+    private static final long DEFAULT_TIMEOUT = 30;
+
+    private static final int DELAY = 30;
+
+    private static final int JOB_TIMEOUT = 300;
+
+    private Environment environment;
+
+    public void runTests() throws InterruptedException, SQLException, JSONException, IOException {
+        runSystemWideExport();
+        runContractNumberExport();
+        testDelete();
+        //stressTest();
+    }
+
+    public TestRunner(Environment environment) throws IOException, InterruptedException, JSONException {
+        this.environment = environment;
+        init();
+    }
+
+    public void init() throws IOException, InterruptedException, JSONException {
+        if(environment.isUsesDockerCompose()) {
+            DockerComposeContainer container = new DockerComposeContainer(
+                    new File("../docker-e2e-compose.yml"))
+                    //.withScaledService("api", 2) // failing now since it's not changing ports
+                    //.withScaledService("worker", 2)
+                    .withExposedService("db", 5432)
+                    .withExposedService("api", 8080);
+            container.start();
+        }
+
+        Yaml yaml = new Yaml();
+        InputStream inputStream = getClass().getResourceAsStream("/" + environment.getConfigName());
+        Map<String, String> yamlMap = yaml.load(inputStream);
+        String oktaUrl = yamlMap.get("okta-url");
+        TEST_USER = yamlMap.get("username");
+        String password = yamlMap.get("password");
+        AB2D_API_URL = yamlMap.get("ab2d-api-url");
+        AB2D_ADMIN_URL = yamlMap.get("ab2d-admin-url");
+
+        String oktaClientId = yamlMap.get("okta-client-id");
+        String oktaPassword = yamlMap.get("okta-password");
+        String authEncoded = Base64.getEncoder().encodeToString((oktaClientId + ":" + oktaPassword).getBytes());
+
+        httpClient = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .build();
+
+        var jwtRequestParms = new HashMap<>() {{
+            put("grant_type", "password");
+            put("username", TEST_USER);
+            put("password", password);
+            put("scope", "openid");
+        }};
```

Comment:
I think you can transition the Okta piece to using client id/secret only, i.e. no human user accounts anymore. Eileen C Frierson will have to go. You can use one of the "PDP-nnnn" accounts. Ping me for help with setting those up.